### PR TITLE
fix: theme generator tokens and recursive resolution

### PR DESCRIPTION
**Pull request by MIA-Deltat1995**
### Description

This PR improves the theme generator algorithm to allow to resolve also theme tokens referring recursively to other theme tokens.

### Checklist

- [x] commit message and branch name follow conventions
- [x] tests are included
- [x] changes are accessible and documented from components stories
- [x] typings are updated or integrated accordingly with your changes
- [x] all added components are exported from index file (if necessary)
- [x] all added files include Apache 2.0 license
- [x] you are not committing extraneous files or sensitive data
- [x] the browser console does not have any logged errors
- [x] necessary labels have been applied to this pull request (enhancement, bug, ecc.)

### CODE REVIEW

File: src/themes/utils/generateTheme.ts [hunk 1]
```diff
@@ -40,17 +40,29 @@ const getFile = (theme: string, file: string): string => resolve(THEMES_DIR, the
...
+
+  const path = nodeValue.replace(PARENTHESES, '').split('.')
+  const { $value } = get(themeValues, path) || {}
 
-    const { $value } = get(themeValues, path) || {}
+  if (!$value) {
+    // The value is referred to another token value instead of a primitive
+    const { $value: tokenValue } = get(themeTokens, path) || {}
+    if (!tokenValue) {
+      throw new Error(`Something goes wrong resolving ${nodeValue}`)
```
Comment (fredmaggiowski) [inline src/themes/utils/generateTheme.ts:59]:
```suggestion
      throw new Error(`Something went wrong resolving ${nodeValue}`)
```

File: src/themes/utils/generateTheme.ts [hunk 2]
```diff
@@ -63,17 +75,29 @@ export default async function generateTheme(themeName: string): Promise<void> {
...
+  const resolveVariable = resolveThemeValues(themeValues, themeStructure)
 
-  const resolvedTheme: Theme = traverse(themeStructure).map(function(node) {
+  return traverse(themeStructure).map(function(node) {
     const isLeaf = node?.$value && node?.$type
-    const resolvedNode = resolveVariable(isLeaf ? node.$value : node)
 
-    this.update(resolvedNode)
+    if (!isLeaf) {
+      this.update(node)
```
Comment (fredmaggiowski) [inline src/themes/utils/generateTheme.ts:93]: why isn't the `resolveVariable` invoked over `node` as in the previous implementation?

Comment (MIA-Deltat1995) [reply src/themes/utils/generateTheme.ts:93]: because there is no value to resolve, but previously the function returned the same node

**Review by fredmaggiowski — APPROVED** [review]
LGTM

**Review by Danielecina — APPROVED** [review]

**Pull request merged**